src/pytorch_lightning/utilities/parsing.py

```diff
@@ -181,13 +185,13 @@ def collect_init_args(
     if not isinstance(frame.f_back, types.FrameType):
         return path_args
 
-    if "__class__" in local_vars:
+    if "__class__" in local_vars and issubclass(local_vars["__class__"], classes):
```

Comment:
Reviewers: This function `collect_init_args` is recursive, it will go up the stack frames until it hits a class's `__init__`. But this init could be inside of another init frame. On master, we would just continue iterating. Here, I'm making sure that we are only doing shenanigans when the init belongs to a subclass of LM or LDM.